packages/next-intl/src/plugin/hasStableTurboConfig.tsx

```diff
@@ -1,8 +1,17 @@
-// eslint-disable-next-line import/order
 import {createRequire} from 'module';
 
-const require = createRequire(import.meta.url);
-const pkg = require('next/package.json');
+function getCurrentVersion() {
+  try {
+    const require = createRequire(import.meta.url);
+    const pkg = require('next/package.json');
+    return pkg.version;
+  } catch (error) {
+    throw new Error(
+      'Failed to get current Next.js version. This can happen if next-intl/plugin is imported outside of your next.config.js, like in your test setup',
```

Comment:
It's true that the error might become visible in a test or e.g. in the Edge runtime (due to missing Node.js APIs). However, I'd say it's less the issue that you're importing this into a test setup, but it's already not ideal if you import this into any of your app code since we're eagerly doing unnecessary work (reading `package.json`).

So I think I'd focus more on that in the error message:

```suggestion
      'Failed to get current Next.js version. This can happen if next-intl/plugin is imported into your app code outside of your next.config.js.,
```

What do you think?